packages/python/plotly/codegen/datatypes.py

```diff
@@ -400,6 +410,19 @@ def __init__(self"""
 """
     )
 
+    if node.name_property in deprecated_mapbox_traces:
+        buffer.write(
+            f"""
+        warn(
+            "*{node.name_property}* is deprecated!"
+            + " Use *{node.name_property.replace("mapbox", "map")}* instead."
+            + " Learn more at: https://plotly.com/python/mapbox-to-maplibre/",
+            stacklevel=2,
+            category=DeprecationWarning,
```

Comment:
@emilykl Good improvement. And I can approve this PR.
As follow ups to this PR:
Could you possibly 
1. double check that `stacklevel=2` is the best option we could use?
2. should we use similar `stacklevel` and `category` options for other warnings in plotly.py?
Thank you!

Reply:
I believe `stacklevel=2` is the correct option in this case -- it causes the warning to appear to be emitted at the place in the user's code where they have called the deprecated function, for example:

```
Making a go.Choroplethmapbox figure...
/Users/ekl/code/plotly.py/my_script.py:32: DeprecationWarning:

*choroplethmapbox* is deprecated! Use *choroplethmap* instead. Learn more at: https://plotly.com/python/mapbox-to-maplibre/
```

(Also, for `DeprecationWarning` specifically, using any other `stacklevel` for these deprecations causes the warning to not appear at all.)

It's just coincidence that `stacklevel=2` is the correct option in this case for both the `go` and `px` deprecations... the correct stacklevel will differ in every case depending on exactly where the warning is being emitted. 

I don't think it's worth the effort to go through the codebase and clean up existing warnings, but I do think it makes sense to use `stacklevel` and `category` consistently going forward. It seems at least some of the warnings already in place in Plotly.py are using them correctly.